internal/podtemplatespec/podtemplatespec.go

```diff
@@ -0,0 +1,119 @@
+package podtemplatespec
```

Comment:
This package should indicate that this is a mutation library on the PodTemplateSpec

```suggestion
package mutatepodtemplatespec
```

Reply:
I don't love my name suggestion.

Reply:
For naming defer to effective go:
> By convention, one-method interfaces are named by the method name plus an -er suffix or similar modification to construct an agent noun: Reader, Writer, Formatter, CloseNotifier etc.

The interface should use the -er naming convention based on the single method interface name. 

controller should define and expect a PodTemplateSpecMutator.
```
type PodTemplateSpecMutator interface {
Mutate(pts corev1.PodTemplateSpec, lri corev1.LimitRangeItem) (corev1.PodTemplateSpec, error)
}

```

Seems like this could move to internal/mutators/podtemplatespec.go or internal/mutators/podtemplatespec/
or a pkg/mutators namespace. 

ApplyResourceRequirements should become something like:

```
package mutators

type pts struct {}

func NewPTSMutator() *pts {
return &pts{}
}

func (p *pts) Mutate(pts corev1.PodTemplateSpec, lri corev1.LimitRangeItem) (corev1.PodTemplateSpec, error) {
...
}
```

Reply:
Moved to `internal/mutators/podtemplatespec.go` to be part of new `mutators` package.
Changed struct to now be empty.
Changed name from `ApplyResourceRequirements` to `Mutate`, and changed function to include pts as input.